static/app/utils/profiling/profile/chromeTraceProfile.tsx

```diff
@@ -285,3 +289,128 @@ export function parseChromeTraceArrayFormat(
     profiles,
   };
 }
+
+function isProfileEvent(event: ChromeTrace.Event): event is ChromeTrace.ProfileEvent {
+  return event.ph === 'P' && event.name === 'Profile';
+}
+
+function isProfileChunk(
+  event: ChromeTrace.Event
+): event is ChromeTrace.ProfileChunkEvent {
+  return event.ph === 'P' && event.name === 'ProfileChunk';
+}
+
+function isThreadmetaData(
+  event: ChromeTrace.Event
+): event is ChromeTrace.ThreadMetadataEvent {
+  event.name === 'Thread';
+
+  return event.ph === 'M' && event.name === 'Thread';
+}
+
+type Required<T> = {
+  [P in keyof T]-?: T[P];
+};
+
+// This mostly follows what speedscope does for the Chrome Trace format, but we do minor adjustments (not sure if they are correct atm),
+// but the protocol format seems out of date and is not well documented, so this is a best effort.
+function collectEventsByProfile(input: ChromeTrace.ArrayFormat): {
+  profiles: Map<string, Required<ChromeTrace.CpuProfile>>;
+  threadNames: Map<string, string>;
+} {
+  const sorted = input.sort(chronologicalSort);
```

Comment:
This isn't guaranteed to be a stable sort, could that be problematic for events with the same timestamps?

Reply:
I think the only possible case where we have same events with same timestamps is when those events happen on different threads. Since we split by process and thread id later, I dont think it's a problem.